### PR TITLE
Springboot 1

### PR DESCRIPTION
**Pull request by yurwing**

### CODE REVIEW

File: src/main/java/com/example/bankservice/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Override
+    public void delete(User user) {
+        userRepository.delete(user);
+    }
+
+    @Override
+    public User getById(Long id) {
+        return userRepository.getOne(id);
```
Comment (vlykhvar) [inline src/main/java/com/example/bankservice/service/impl/UserServiceImpl.java]: you should .orElseThrow() here.

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/com/example/bankservice/repository/RoleRepository.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.bankservice.repository;
+
+import com.example.bankservice.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query(value = "from Role where roleName = ?1")
+    Optional<Role> getByName(Role.RoleName name);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankservice/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/com/example/bankservice/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.example.bankservice.service;
+
+import com.example.bankservice.model.User;
+
+public interface UserService {
+    User save(User user);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankservice/service/UserService.java]: see checklist

File: src/main/java/com/example/bankservice/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankservice/service/impl/UserServiceImpl.java]: let's call the method that join fetches roles

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let’s configure access to h2 database through the console in `application.properties`

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**